### PR TITLE
Wording updates for the events tab

### PR DESCRIPTION
**Pull request by viroulep**
I've updated the mbf time limit to include the maximum allowed time, and the events format to include the format for all phases when displaying a combined round.

The format column can be a little long with combined round: I introduced a "short name" alternative for formats (with a tooltip containing the format's full name), which could be used to save some space.
Not sure how good an idea this is, so I'm gonna try to have feedback before going forward with this.

### CODE REVIEW

File: WcaOnRails/app/views/competitions/_events_tab.html.erb [hunk 1]
```diff
@@ -18,7 +18,14 @@
             <%= competition_event.event.name if round.number == 1 %>
           </td>
           <td><%= round.round_type.name %></td>
-          <td><%= round.format.name %></td>
+          <% round_cutoff = round.cutoff %>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/competitions/_events_tab.html.erb]: Any reason for making a variable? It's of the same length as `round.cutoff` anyway ^^

File: WcaOnRails/config/locales/en.yml [hunk 2]
```diff
@@ -68,6 +68,13 @@ en:
     "3": "Best of 3"
     "a": "Average of 5"
     "m": "Mean of 3"
+    #context: short name alternative
+    short:
+      "1": "Bo1"
+      "2": "Bo2"
+      "3": "Bo3"
+      "a": "Ao5"
+      "m": "Mo3"
```
Comment (jonatanklosko) [inline WcaOnRails/config/locales/en.yml]: Do you think it's likely to differ across languages? Just pointing out that when updating Polish translation I kept the full names in English - they are more popular anyway and there's no way of differentiating between *mean* and *average* using a single world in Polish. That's why I wonder if there's any actual chance of short formats to differ.

**Comment by pedrosino** [comment]
Maybe a screenshot? :D

File: WcaOnRails/config/locales/en.yml (hunk 2 above)
Comment (viroulep) [reply WcaOnRails/config/locales/en.yml]: I actually don't know :s
Long version of formats may surely change across language, but for short names I was also thinking of using the English ones...

File: WcaOnRails/app/views/competitions/_events_tab.html.erb (hunk 1 above)
Comment (viroulep) [reply WcaOnRails/app/views/competitions/_events_tab.html.erb]: Nop.

File: WcaOnRails/config/locales/en.yml [hunk 3]
```diff
@@ -1111,6 +1118,7 @@ en:
         cumulative_one_round_html: "A %{cumulative_time_limit} may be enforced (see %{regulation_link})."
         cumulative_across_rounds_html: "A %{cumulative_time_limit} may be enforced across rounds (see %{guideline_link})."
         cutoff_html: "The result to beat to proceed to the second phase of a combined round (see %{regulation_link})."
+        format_html: "The format describes how to determine the ranking of competitors based on their results. The list of allowed formats per event is described in %{link_to_9b}. See %{link_to_9f} for a description of each format."
```
Comment (viroulep) [inline WcaOnRails/config/locales/en.yml]: @thewca/wrc-team any input on the wording of the format description?

**Comment by viroulep** [comment]
Screenshots with "short" names and full names (using the French translation which is pretty long):
short_names

names

File: WcaOnRails/app/models/competition.rb [hunk 4]
```diff
@@ -696,6 +696,11 @@ def in_progress?
     !results_posted? && (start_date..end_date).cover?(Date.today)
   end
 
+  def uses_cutoff?
+    cutoff = competition_events.find { |ce| ce.rounds.find { |r| r.cutoff } }
+    !cutoff.nil?
+  end
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/competition.rb]: Using `any?` would directly result in a boolean value =)
```ruby
def uses_cutoff?
  competition_events.any? { |ce| ce.rounds.any?(&:cutoff) }
end
```

Comment (viroulep) [reply WcaOnRails/app/models/competition.rb]: Nice :)

File: WcaOnRails/app/models/round.rb [hunk 5]
```diff
@@ -56,6 +56,23 @@ def round_type_id
...
+    phase_formats.map! do |f|
+      if with_tooltips
+        content_tag(:span, f.short_name, data: { toggle: "tooltip" }, title: f.name)
+      elsif with_short_names
+        f.short_name
+      else
+        f.name
+      end
+    end
+    cutoff_format ? safe_join(phase_formats, " / ") : phase_formats[0]
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/round.rb]: Wouldn't `safe_join(phase_formats, " / ")` suffice?

Comment (viroulep) [reply WcaOnRails/app/models/round.rb]: Yes it does!

**Pull request merged**